packages/example/src/components/Notifications/History.tsx

```diff
@@ -0,0 +1,201 @@
+import type { Notification } from '@usedapp/core'
+import { ChainId, useNotificationsContext, useTransactionsContext } from '@usedapp/core'
+import type { ReactNode } from 'react'
+import styled from 'styled-components'
+import { TextBold } from '../../typography/Text'
+import { ContentBlock } from '../base/base'
+
+interface TableWrapperProps {
+  children: ReactNode
+  title: string
+}
+
+const TableWrapper = ({ children, title }: TableWrapperProps) => (
+  <ContentBlock>
+    <CellTitle>{title}</CellTitle>
+    <Table>{children}</Table>
+  </ContentBlock>
+)
+
+interface TransactionListProps {
+  chainId: ChainId
+}
+
+export const TransactionsList = ({ chainId }: TransactionListProps) => {
+  const { transactions } = useTransactionsContext()
+  const chainTransactions = transactions[chainId] ?? []
+  const reversed = [...chainTransactions].reverse()
+
+  return (
+    <TableWrapper title="Transactions history">
+      {reversed.map((tx) => (
+        <div key={tx.transaction.hash}>
+          {!tx.receipt && <Spinner />}
+          <span>{tx.transaction.hash.slice(0, 6) + '...' + tx.transaction.hash.slice(-4)}</span>
+        </div>
+      ))}
+    </TableWrapper>
+  )
+}
+
+interface NotificationPanelProps {
+  title: string
+  icon: ReactNode
+  transactionHash: string
+}
+
+const NotificationPanel = ({ title, icon, transactionHash }: NotificationPanelProps) => {
+  function getEtherscanLink(hash: string) {
+    return `https://kovan.etherscan.io/tx/${hash}`
+  }
+  return (
+    <NotificationWrapper>
+      <IconContainer>{icon}</IconContainer>
+      <NotificationDetailsWrapper>
+        <TextBold>{title}</TextBold>
+        <Link href={getEtherscanLink(transactionHash)} target="_blank" rel="noopener noreferrer">
+          View on Etherscan
+        </Link>
+      </NotificationDetailsWrapper>
+    </NotificationWrapper>
+  )
+}
+
+interface NotificationItemProps {
+  notification: Notification
+}
+
+const NotificationItem = ({ notification }: NotificationItemProps) => {
+  switch (notification.type) {
```

Comment:
You should be able to do a switch/mapping just on the icon, i.e.
```ts
const label = {'transactionStarted': "Transaction started", ....}
...

function NotificationIcon() {
  switch(type) {
    case 'transactionStarted': return (<ClockIcon />)
    ...
  }
}
```

Reply:
I don't know if I understand this correctly, but anyway I need to check the type of transaction just to get hash from it.
The only thing that I can get from notification by default right now is `type`.